dali/operators/sequence/optical_flow/turing_of/utils.h

```diff
@@ -15,15 +15,79 @@
-        NV_OF_STATUS errorCode = nvOFAPI;                                            \
-        if (errorCode != NV_OF_SUCCESS) {                                            \
-            std::ostringstream errorLog;                                             \
-            errorLog << #nvOFAPI << " returned error: " << errorCode << std::endl;   \
-            DALI_FAIL(errorLog.str());                                               \
-        }                                                                            \
-    } while (0)
+#include <string>
+#include "dali/core/cuda_error.h"
+#include "dali/core/format.h"
+
+namespace dali {
+
+class NvofError : public std::runtime_error {
+ public:
+  explicit NvofError(NV_OF_STATUS result, const char *details = nullptr)
+  : std::runtime_error(Message(result, details))
+  , result_(result) {}
+
+  static const char *ErrorString(NV_OF_STATUS result) {
+    switch (result) {
+      case NV_OF_SUCCESS:
+        return "Nvidia Optical flow operation was successful";
+      case NV_OF_ERR_OF_NOT_AVAILABLE:
+        return "Nvidia HW Optical flow functionality is not supported";
+      case NV_OF_ERR_UNSUPPORTED_DEVICE:
+        return "The device passed by the client to Nvidia HW Optical flow is not supported";
+      case NV_OF_ERR_DEVICE_DOES_NOT_EXIST:
+        return "The device passed to the Nvidia HW Optical flow API call is no longer available "
+                "and needs to be reinitialized";
+      case NV_OF_ERR_INVALID_PTR:
+        return "One or more of the pointers passed to the Nvidia HW Optical flow API "
+               "API call is invalid";
+      case NV_OF_ERR_INVALID_PARAM:
+        return "One or more of the parameter passed to the Nvidia Optical flow API call is invalid";
+      case NV_OF_ERR_INVALID_CALL:
+        return "Nvidia Optical flow API call was made in wrong sequence/order";
+      case NV_OF_ERR_INVALID_VERSION:
+        return "An invalid struct version was used by the Nvidia Optical flow client";
+      case NV_OF_ERR_OUT_OF_MEMORY:
+        return "Nvidia Optical flow API call failed because it was unable to allocate  "
+               "enough memory to perform the requested operation";
+      case NV_OF_ERR_NOT_INITIALIZED:
+        return "Nvidia Optical flow session has not been initialized with or initialization "
+               "has failed";
+      case NV_OF_ERR_UNSUPPORTED_FEATURE:
+        return "Unsupported parameter was passed by the client to Nvidia Optical flow";
+      case NV_OF_ERR_GENERIC:
+        // fallthrough
+      default:
+        return "< unknown error >";
+    }
+  }
+
+  static std::string Message(NV_OF_STATUS result, const char *details) {
+    if (details && *details) {
+      return make_string("Nvidia Optical flow error: ", result, " ", ErrorString(result),
+                         "\nDetails:\n", details);
+    } else {
+      return make_string("Nvidia Optical flow error: ", result, " ", ErrorString(result));
+    }
```

Comment:
General remark, shouldn't "Nvidia" word in these error messages be capitalized? Up to you, just a suggestion
```suggestion
    if (details && *details) {
      return make_string("NVIDIA Optical flow error: ", result, " ", ErrorString(result),
                         "\nDetails:\n", details);
    } else {
      return make_string("NVIDIA Optical flow error: ", result, " ", ErrorString(result));
    }
```

Reply:
Done